cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/AbstractSchedulerService.java

```diff
@@ -280,9 +293,13 @@ public ScheduleState scheduleState(String scheduleId) {
       }
     }
 
-    private String getJobKey(Id.Program program, ProgramType programType) {
-      return String.format("%s:%s:%s:%s", programType.name(), program.getAccountId(),
-                           program.getApplicationId(), program.getId());
+    private void checkInitialized() {
+      Preconditions.checkNotNull(scheduler, "Scheduler not yet initialized");
+    }
+
+    private JobKey getJobKey(Id.Program program, ProgramType programType) {
+      return new JobKey(String.format("%s:%s:%s:%s", programType.name(), program.getAccountId(),
```

Comment:
I see the change in the key format. However there is no corresponding test case changes. Can we add a testcase if it is not there? If there is a test case then we would need to update them.